pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -518,10 +520,15 @@ func (r *ReconcileBareMetalHost) actionProvisioning(prov provisioner.Provisioner
 		}
 		err = r.client.Get(context.TODO(), key, userDataSecret)
 		if err != nil {
-			return "", errors.Wrap(err,
+			return "", nil, errors.Wrap(err,
 				"failed to fetch user data from secret reference")
 		}
-		return string(userDataSecret.Data["userData"]), nil
+		var networkData map[string]interface{}
+		if err = yaml.Unmarshal(userDataSecret.Data["networkData"], &networkData); err != nil {
+			return "", nil, errors.Wrap(err, "failed to econde network_data.json from secret")
+		}
+
+		return string(userDataSecret.Data["userData"]), networkData, nil
```

Comment:
We should probably also harden this in case the userData key isn't present, but that can be a separate PR.

Reply:
done